llvm/lib/Target/SPIRV/SPIRVEmitIntrinsics.cpp

```diff
@@ -1074,6 +1084,10 @@ void SPIRVEmitIntrinsics::processGlobalValue(GlobalVariable &GV,
 
 void SPIRVEmitIntrinsics::insertAssignPtrTypeIntrs(Instruction *I,
                                                    IRBuilder<> &B) {
+  // Don't assign types to LLVM tokens.
+  if (isConvergenceIntrinsic(I))
```

Comment:
Maybe move this check to SPIRVEmitIntrinsics::runOnFunction(), where insertAssignPtrTypeIntrs() is called.

Reply:
Yes, that's better, changed, thanks!